packages/openapi-v2/test/unit/controller-spec/controller-decorators/param-decorators/param.test.ts

```diff
@@ -270,6 +278,98 @@ describe('Routing metadata for parameters', () => {
       expect(actualSpec.paths['/greet']['get']).to.eql(expectedSpec);
     });
 
+    it('infers complex body parameter definition into the controller spec', () => {
+      const fooSpec: ParameterObject = {
+        name: 'foo',
+        in: 'body',
+      };
+      const barSpec: ParameterObject = {
+        name: 'bar',
+        in: 'body',
+      };
+      @model()
+      class MySoul {
+        @property() price: number;
+      }
+      @model()
+      class MyBody {
+        @property() name: string;
+        @property() soul: MySoul;
+      }
+      class MyController {
+        @post('/foo')
+        foo(@param(fooSpec) foo: MyBody) {}
+
+        @post('/bar')
+        bar(@param(barSpec) bar: MySoul) {}
+      }
+      const expectedDefs = {
+        definitions: {
+          MyBody: {
+            properties: {
+              name: {
+                type: 'string',
+              },
+              soul: {
+                $ref: '#definitions/MySoul',
+              },
+            },
+          },
+          MySoul: {
```

Comment:
I think this part is redundant - we have already verified elsewhere that "simple" models are correctly converted to JSON schema. Please simplify this test so that it asserts only the important/relevant facts.

For example:

```ts
expect(Object.keys(spec.definitions)).to.deepEqual(['MyBody', 'MySoul']);
expect(spec.definitions.MyBody).to.deepEqual({
  // ...
});
```